contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -207,9 +249,27 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
      * Internal function without access restriction.
      */
     function _beginDefaultAdminTransfer(address newAdmin) internal virtual {
-        _defaultAdminTransferDelayedUntil = SafeCast.toUint48(block.timestamp) + defaultAdminDelay();
+        (, uint48 delaySchedule) = pendingDefaultAdminDelay();
+        if (_isSet(delaySchedule)) _rollbackDefaultAdminDelay();
+
+        _pendingDefaultAdminSchedule = SafeCast.toUint48(block.timestamp) + defaultAdminDelay();
         _pendingDefaultAdmin = newAdmin;
-        emit DefaultAdminRoleChangeStarted(pendingDefaultAdmin(), defaultAdminTransferDelayedUntil());
+
+        (address newDefaultAdmin, uint48 transferSchedule) = pendingDefaultAdmin();
+        emit DefaultAdminTransferScheduled(newDefaultAdmin, transferSchedule);
```

Comment:
IMO this is taking virtual functions too far. We've just set the values, here we should use the values we just set. If someone is overriding this function, it's on them to make sure their override is consistent with this logic.

```suggestion
        emit DefaultAdminTransferScheduled(newAdmin, _pendingDefaultAdminSchedule);
```

Reply:
Yes, I also realized by taking a look again for the docs. Just updated.